xds/src/test/java/io/grpc/xds/ClientXdsClientDataTest.java

```diff
@@ -773,245 +803,197 @@ public void parseCluster_ringHashLbPolicy_invalidRingSizeConfig_tooLargeRingSize
     Filter filter =
-        buildHttpConnectionManager(
-            "envoy.http_connection_manager",
+        buildHttpConnectionManagerFilter(
             HttpFilter.newBuilder().setName("hf").setIsOptional(true).build(),
             HttpFilter.newBuilder().setName("hf").setIsOptional(true).build());
-    FilterChain filterChain = buildFilterChain(filter);
-    Listener listener =
-        Listener.newBuilder()
-            .setName("listener1")
-            .setTrafficDirection(TrafficDirection.INBOUND)
-            .addFilterChains(filterChain)
-            .build();
-    StructOrError<io.grpc.xds.EnvoyServerProtoData.Listener> struct =
-        ClientXdsClient.parseServerSideListener(listener, null);
-    assertThat(struct.getErrorDetail())
-        .isEqualTo("http-connection-manager has non-unique http-filter name:hf");
-  }
-
-  @Test
-  public void parseServerSideListener_configDiscoveryHttpFilter() {
-    Filter filter =
-        buildHttpConnectionManager(
-            "envoy.http_connection_manager",
-            HttpFilter.newBuilder()
-                .setName("envoy.router")
-                .setConfigDiscovery(ExtensionConfigSource.newBuilder().build())
-                .build());
-    FilterChain filterChain = buildFilterChain(filter);
-    Listener listener =
-        Listener.newBuilder()
-            .setName("listener1")
-            .setTrafficDirection(TrafficDirection.INBOUND)
-            .addFilterChains(filterChain)
-            .build();
-    StructOrError<io.grpc.xds.EnvoyServerProtoData.Listener> struct =
-        ClientXdsClient.parseServerSideListener(listener, null);
-    assertThat(struct.getErrorDetail())
-        .isEqualTo(
-            "http-connection-manager http-filter envoy.router uses "
-                + "config-discovery which is unsupported");
-  }
-
-  @Test
-  public void parseServerSideListener_badTypeUrlHttpFilter() {
-    HTTPFault fault = HTTPFault.newBuilder().build();
-    Filter filter =
-        buildHttpConnectionManager(
-            "envoy.http_connection_manager",
-            HttpFilter.newBuilder()
-                .setName("envoy.router")
-                .setTypedConfig(Any.pack(fault, "type.googleapis.com"))
-                .build());
-    FilterChain filterChain = buildFilterChain(filter);
-    Listener listener =
-        Listener.newBuilder()
-            .setName("listener1")
-            .setTrafficDirection(TrafficDirection.INBOUND)
-            .addFilterChains(filterChain)
-            .build();
-    StructOrError<io.grpc.xds.EnvoyServerProtoData.Listener> struct =
-        ClientXdsClient.parseServerSideListener(listener, null);
-    assertThat(struct.getErrorDetail())
-        .isEqualTo(
-            "http-connection-manager http-filter envoy.router has unsupported typed-config type:"
-                + "type.googleapis.com/envoy.extensions.filters.http.fault.v3.HTTPFault");
-  }
-
-  @Test
-  public void parseServerSideListener_missingTypeUrlHttpFilter() {
```

Comment:
This too. If we have made the conscious decision to remove these tests because they don't seem important let me know

Reply:
Same see `parseHttpFilter_unsupportedButOptional()` and `parseHttpFilter_unsupportedAndRequired()`.